src/components/UserMasterPanel.js

```diff
@@ -56,36 +55,49 @@ const UserMasterPanel = (props) => {
           onChange={(lastName) => onEditedChanged({ ...edited, lastName })}
         />
       </Grid>
-      {!(obligatory_user_fields?.email == 'H' || edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.email == 'H') &&
-            <Grid item xs={4} className={classes.item}>
-            <TextInput
-              module="admin"
-              type="email"
-              label="user.email"
-              required = {obligatory_user_fields?.email == 'M' || (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.email == 'M')}
-              readOnly={readOnly}
-              value={edited?.email ?? ""}
-              onChange={(email) => onEditedChanged({ ...edited, email })}
-            />
-          </Grid>
-      }
-      {!(obligatory_user_fields?.phone == 'H' || edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.phone == 'H') && 
+      {!(
+        obligatory_user_fields?.email == "H" ||
+        (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.email == "H")
+      ) && (
+        <Grid item xs={4} className={classes.item}>
+          <TextInput
+            module="admin"
+            type="email"
+            label="user.email"
+            required={
+              obligatory_user_fields?.email == "M" ||
+              (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.email == "M")
+            }
+            readOnly={readOnly}
+            value={edited?.email ?? ""}
+            onChange={(email) => onEditedChanged({ ...edited, email })}
+          />
+        </Grid>
+      )}
+      {!(
+        obligatory_user_fields?.phone == "H" ||
+        (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.phone == "H")
+      ) && (
         <Grid item xs={4} className={classes.item}>
           <TextInput
             module="admin"
             type="phone"
             label="user.phone"
-            required = {obligatory_user_fields?.phone == 'M' || (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.phone == 'M')}
+            required={
+              obligatory_user_fields?.phone == "M" ||
+              (edited.userTypes?.includes(ENROLMENT_OFFICER_USER_TYPE) && obligatory_eo_fields?.phone == "M")
```

Comment:
In the future we could add enum fot those 'M'/'H'. But I know that currently is not possible to change this.

Reply:
Good point, it should be changed.